modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1013,11 +1014,21 @@ private ObjectRelationship _updateObjectRelationship(
 	}
 
 	private void _validate(
-			String externalReferenceCode, long objectDefinitionId1,
-			long objectDefinitionId2, long parameterObjectFieldId, String name,
-			String type)
+			String externalReferenceCode, long companyId,
+			long objectDefinitionId1, long objectDefinitionId2,
+			long parameterObjectFieldId, String name, String type)
 		throws PortalException {
 
+		if (Validator.isNotNull(externalReferenceCode)) {
+			ObjectRelationship objectRelationship =
+				objectRelationshipPersistence.fetchByERC_C_ODI1(
+					externalReferenceCode, companyId, objectDefinitionId1);
+
+			if (objectRelationship != null) {
+				throw new DuplicateObjectRelationshipExternalReferenceCodeException();
+			}
+		}
+
```

Comment:
Could you create a separated method only to validate the ERC, please?
Like: _validateExternalReferenceCode()
Murilo refactored this validate method and separated the validation about the properties `name` and `type` (https://github.com/liferay-objects/liferay-portal/pull/2322);

Reply:
When creating this method, could you please add the failure first like this:
```
if (Validator.isNull(externalReferenceCode)) {
	return;
}
```